sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Listeners/ServiceBusScalerProvider.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Azure.Messaging.ServiceBus;
+using Azure.Messaging.ServiceBus.Administration;
+using Microsoft.Azure.WebJobs.Extensions.ServiceBus.Config;
+using Microsoft.Azure.WebJobs.Host.Scale;
+using Microsoft.Azure.WebJobs.ServiceBus;
+using Microsoft.Azure.WebJobs.ServiceBus.Listeners;
+using Microsoft.Extensions.Azure;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Azure.WebJobs.Extensions.ServiceBus.Listeners
+{
+    internal class ServiceBusScalerProvider : IScaleMonitorProvider, ITargetScalerProvider
+    {
+        private readonly IConfiguration _configuration;
+        private AzureComponentFactory _defaultAzureComponentFactory;
+        private readonly MessagingProvider _messagingProvider;
+        private readonly IOptions<ServiceBusOptions> _options;
+        private readonly AzureEventSourceLogForwarder _logForwarder;
+        private readonly TriggerMetadata _triggerMetadata;
+        private readonly ILoggerFactory _loggerFactory;
+
+        public ServiceBusScalerProvider(IServiceProvider serviceProvider, TriggerMetadata triggerMetadata)
+        {
+            _configuration = serviceProvider.GetService<IConfiguration>();
+            _defaultAzureComponentFactory = triggerMetadata.Properties[nameof(AzureComponentFactory)] as AzureComponentFactory;
+            _messagingProvider = serviceProvider.GetService<MessagingProvider>();
+            _triggerMetadata = triggerMetadata;
+            _logForwarder = serviceProvider.GetService<AzureEventSourceLogForwarder>();
+            _options = serviceProvider.GetService<IOptions<ServiceBusOptions>>();
+            _loggerFactory = serviceProvider.GetService<ILoggerFactory>();
+        }
+
+        public IScaleMonitor GetMonitor()
+        {
+            ServiceBusMetadata serviceBusMetadata = JsonConvert.DeserializeObject<ServiceBusMetadata>(_triggerMetadata.Metadata.ToString());
+            AzureComponentFactory azureComponentFactory = GetAzureComponentFactory(_triggerMetadata);
+
+            (ServiceBusAdministrationClient adminClient, ServiceBusReceiver receiver) = CreateParameters(_configuration, serviceBusMetadata, azureComponentFactory);
+
+            return new ServiceBusScaleMonitor(
+                _triggerMetadata.FunctionName,
```

Comment:
Is this the same ID value as in other places where we create a monitor, e.g. https://github.com/Azure/azure-sdk-for-net/blob/460129360094a5617743024ec3e369055ad624fe/sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Listeners/ServiceBusListener.cs#L107? I think ID is the full name?

Reply:
The value will be different for RDS and SC scenarios:
RDS - Microsoft.Azure.WebJobs.Host.EndToEndTests.ServiceBusEndToEndTests+ServiceBusTestJobs.SBQueue2SBQueue (full name)
SC - SBQueue2SBQueue (from sync trigger content)

I think it's ok as we use the fucntion id as a key for storage base metric repository in `TableStorageScaleMetricsRepository`